rails/app/views/dashboard/stories/_stories.html.erb

```diff
@@ -1,10 +1,38 @@
 <% @stories.each do |story| %>
   <%= link_to story, class: "card" do %>
     <div>
-      <h3>
-        <span class="card__heading--small"><%= story.topic %></span>
-        <%= story.title %>
-      </h3>
+      <div class="story-header">
+        <div>
+          <h3 style="width:max-content">
+            <span class="card__heading--small"><%= story.topic %></span>
+            <%= story.title %>
+          </h3>
+        </div>
+        <% if current_user.member? && story.story_pinned %>
+          <div class="view-icon-pinned">
+            <svg class="icon-pinned"><use href="#icon-thumbtack"></svg>
+          </div>
+        <% end %>
+        <% if current_user.admin? || current_user.editor? %>
+          <% if story.story_pinned %>
+            <div class="edit-icon-pinned">
+              <%= form_with model: @story, url: unpin_story_path(story), method: "patch", multipart: true, class: "form", data: {remote: false}, locale: true do |f| %>
+                <%= f.button nil, class: "icon-button" do %>
+                  <svg class="icon-pinned"><use href="#icon-thumbtack"></svg>
+                <% end %>
+              <% end %>
+            </div>
+          <% else %>
+            <div class="edit-icon-unpinned">
+              <%= form_with model: @story, url: pin_story_path(story), method: "patch", multipart: true, class: "form", data: {remote: false}, locale: true do |f| %>
+                <%= f.button nil, class: "icon-button" do %>
+                  <svg class="icon-unpinned"><use href="#icon-thumbtack"></svg>
+                <% end %>
+              <% end %>
+            </div>
+          <% end %>
```

Comment:
I think this could benefit from using Rails' [`button_to` helper](https://api.rubyonrails.org/v7.1.1/classes/ActionView/Helpers/UrlHelper.html#method-i-button_to). It essentially creates the code you're doing here, but streamlines the form creation.

```suggestion
          <% if story.story_pinned %>
            <%= button_to nil, unpin_story_path(story), method: "patch", class: "edit-icon-pinned", data: {remote: false}, locale: true  do %>
              <svg class="icon-pinned"><use href="#icon-thumbtack"></svg>
            <% end %>
          <% else %>
            <%= button_to nil, pin_story_path(story), method: "patch", class: "edit-icon-unpinned", data: {remote: false}, locale: true do %>
              <svg class="icon-unpinned"><use href="#icon-thumbtack"></svg>
            <% end %>
          <% end %>
```

Reply:
NIce idea, I updated here